pkg/evaluator/evaluator.go

```diff
@@ -718,14 +719,29 @@ func evalBinaryBoolExpr(op parser.Operator, left, right *boolVal) (value, error)
 	return nil, fmt.Errorf("%w (bool): %v", ErrOperation, op.String())
 }
 
-func evalBinaryArrayExpr(op parser.Operator, left, right *arrayVal) (value, error) {
-	if op != parser.OP_PLUS {
+func evalBinaryArrayExpr(op parser.Operator, left *arrayVal, right value) (value, error) {
+	switch op {
+	case parser.OP_PLUS:
+		rightArray := right.(*arrayVal)
+		result := left.Copy()
+		rightElements := *rightArray.Copy().Elements
+		*result.Elements = append(*result.Elements, rightElements...)
+		return result, nil
+	case parser.OP_ASTERISK:
+		rightNum := right.(*numVal)
+		repetitions := int(rightNum.V)
+		if float64(repetitions) != float64(rightNum.V) {
+			return nil, fmt.Errorf("%w: %v", ErrBadRepetition, right)
```

Comment:
Maybe this should also include an error case for negative integers? (and maybe deserves a test?)

Reply:
yes it should. good catch. a negative value will crash things I expect

Reply:
done for both evaluator and vm